### PR TITLE
test: Check that calling walletpasshprase does not freeze the node

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

File: test/functional/wallet_encryption.py [hunk 1]
```diff
@@ -38,6 +40,12 @@ def run_test(self):
         assert_raises_rpc_error(-8, "passphrase can not be empty", self.nodes[0].encryptwallet, '')
         self.nodes[0].encryptwallet(passphrase)
 
+        for i in range(5):
+            r = random.uniform(0.9, 1.1)
+            self.log.info('Check that calling walletpasshprase does not freeze the node. sleep={}'.format(r))
```
Comment (jonatack) [inline test/functional/wallet_encryption.py:45]: s/walletpasshprase/walletpassphrase/

**Review by jonatack — COMMENTED** [review]
Good idea. Currently timing out after 60 seconds but I suppose this is for testing #18487.

**Comment by jonatack** [comment]
This passes on #18487.

**Comment by jonatack** [comment]
This passes on #18487, but am seeing another error about 20% of the time:
```
2020-04-02T16:05:38.667000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/test_framework.py", line 112, in main
    self.run_test()
  File "test/functional/wallet_encryption.py", line 131, in run_test
    assert_raises_rpc_error(-13, "Please enter the wallet passphrase with walletpassphrase first", self.nodes[0].dumpprivkey, address)
  File "/home/jon/projects/bitcoin/bitcoin/test/functional/test_framework/util.py", line 114, in assert_raises_rpc_error
    assert try_rpc(code, message, fun, *args, **kwds), "No exception raised"
AssertionError: No exception raised
```

**Comment by promag** [comment]
I don't think this would always fail without #18487 so it doesn't prevent a regression. Maybe we should unit test that RPCRunLater waits for event callback to finish if the call was already triggered?

**Pull request closed without merging**